### PR TITLE
feat(misconf): Add support for independently enabling libraries

### PR DESCRIPTION
**Pull request by simar7**
## Description

Relevant issue: https://github.com/aquasecurity/trivy/issues/4181

A user might need to pass in a custom policy that uses a rego library (e.g. https://github.com/aquasecurity/defsec/blob/master/rules/kubernetes/lib/kubernetes.rego) that we provide. Today it is not possible to use a custom policy like such without providing both the policy and the library.

Signed-off-by: Simar <simar@linux.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,7 +14,7 @@ require (
 	github.com/NYTimes/gziphandler v1.1.1
 	github.com/alicebob/miniredis/v2 v2.30.1
 	github.com/aquasecurity/bolt-fixtures v0.0.0-20200903104109-d34e7f983986
-	github.com/aquasecurity/defsec v0.85.0
+	github.com/aquasecurity/defsec v0.85.1-0.20230414202454-d885472b0fd7
```
Comment (simar7) [inline go.mod:17]: note: Update to latest release when https://github.com/aquasecurity/defsec/pull/1280 is merged and defsec released.

**Comment by itaysk** [comment]
can you please elaborate on the description of this pr? and if applicable create an issue (in trivy) that describes the value to the user?

**Comment by simar7** [comment]
> can you please elaborate on the description of this pr? and if applicable create an issue (in trivy) that describes the value to the user?

Sure I have updated the description. It's a chore not a feature so it adds no value to the user.

**Comment by simar7** [comment]
@knqyf263 could you review it?

**Comment by itaysk** [comment]
do I get it right that it's changing the CLI UX (flags)? If that's the case can you please create a feature issue with justification so we properly track and communicate the change?

**Comment by simar7** [comment]
> do I get it right that it's changing the CLI UX (flags)? If that's the case can you please create a feature issue with justification so we properly track and communicate the change?

There's no CLI UX change (flags). The behaviour remains unchanged, it's just that now scanners are independently able to control enabling policies and libraries. Just breaking one option (that was overloaded) into two for defsec.

**Comment by simar7** [comment]
Marking as draft as I need to get back to this.

**Comment by simar7** [comment]
Related defsec PR: https://github.com/aquasecurity/defsec/pull/1357

**Comment by simar7** [comment]
Will be moved to ready when https://github.com/aquasecurity/defsec/pull/1357 is merged (after v0.90.0 defsec release).

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**